storage/interceptable_repository.go

```diff
@@ -190,6 +190,9 @@ func (ir *interceptableRepository) Update(ctx context.Context, obj types.Object,
 			return nil, err
 		}
 
+		labels, _, _ := query.ApplyLabelChangesToLabels(labelChanges, newObj.GetLabels())
```

Comment:
this is strange. You update the object and receive an updated object as a result, but you still have to update the labels of the object (even though the labels in the db is updated)

Reply:
coz interceptors can add label changes while they are executing